### PR TITLE
start: docs for "head & metadata" and other improvements

### PR DESCRIPTION
**Pull request by nksaraf**

### CODE REVIEW

File: content/start/core-concepts/head-and-metadata.mdx [hunk 1]
```diff
@@ -1,17 +1,178 @@
-<Title>Head and metadata</Title>
+import { Aside } from "~/components/FrameworkAside";
 
-## Populating the `<head>` of your document
+# Head and metadata
 
-### Add tags in `root.tsx`
+Normally when we are building UIs for our apps, we are dealing with DOM elements that are going to be rendered in the `body` of the `document`. But there are cases where we want to customize what's going to be rendered in the `head` of the `document`. The common elements used in the `head` are:
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:5]: I think this is cleaner?
```suggestion
When we are building UIs for our apps, we are normally dealing with DOM elements that are going to be rendered in the `body` of the `document`. But there are cases where we want to customize what's going to be rendered in the `head` of the `document`. The common elements used in the `head` are:
```

File: content/start/core-concepts/head-and-metadata.mdx [hunk 2]
```diff
@@ -1,17 +1,178 @@
...
 
-### Use `solid-meta` inside component tree
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:7]: Is it correct to say `solid-meta` when we are really importing from `solid-start`?

File: content/start/core-concepts/head-and-metadata.mdx [hunk 3]
```diff
@@ -1,17 +1,178 @@
...
+- `title`: Specifies the title of the page, used by the browser tab and headings of search results,
+- `meta`: Specifies a variety of metadata about the page specified by `name`, ranging from favicon, character set to OG tags for SEO.
+- `link`: Adds assets like stylesheets or scripts for the browser to load for the page
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:9]: We don't add scripts via links

```suggestion
- `link`: Adds assets like stylesheets for the browser to load for the page
```

File: content/start/core-concepts/head-and-metadata.mdx [hunk 4]
```diff
@@ -1,17 +1,178 @@
...
-  - Meta
-    - solid-meta on the server (e.g., title tag)
-    - populate `head` section dynamically with `solid-meta`
-  - Links
-    - prefetch CSS in head on per-route basis (prod only)
-  - Scripts
-    - enables JS (links to entry client file). Removing it disables JS in your app.
+### Adding `head` tags
+
+In SolidStart, we ship with actual components representing these tags. So: `title` -> `Title`, `meta` -> `Meta`, etc. They can be used, not only as children of `Head` in `root.tsx`, but anywhere in your app, even deep in your component tree. Powered by `@solidjs/meta`, we can interpret these tags and update the `document.head` appropriately.
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:14]:
```suggestion
SolidStart, has actual components that represent these tags. So: `title` -> `Title`, `meta` -> `Meta`, etc. They can be used, not only as children of `Head` in `root.tsx`, but anywhere in your app, even deep in your component tree. Powered by `@solidjs/meta`, we can interpret these tags and update the `document.head` appropriately.
```

File: content/start/core-concepts/head-and-metadata.mdx [hunk 5]
```diff
@@ -1,17 +1,178 @@
...
+- Adding a default fallback `title` to be used by all pages (if they don't specify their own),
+- Adding common `meta` tags like `charset` and `viewport` for consistent behavior across the app
+- Adding links to external stylesheets, fonts, manifests, favicon
+
+See an example below:
+
+```tsx {9-13}
+import { Suspense } from "solid-js";
+import {
+  ...
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:29]: This isn't a good idea, at least show the head related imports

File: content/start/core-concepts/head-and-metadata.mdx [hunk 6]
```diff
@@ -1,17 +1,178 @@
...
+  For `Title` and `Meta` tags with the same name as one mounted higher up in the
+  tree, the instance that is lower in the tree takes precedence. This means that
+  the `Title` component used inside a Route component will override the `Title`
+  component used in the `Head` component. If you go to a route without a `Title`
+  component, the `Head` component's `Title` component will be used.
+</Aside>
+
+### Adding a site-suffix in Title
+
+You can create custom components that wrap `Title` to add a site-specific prefix to all the titles, eg.
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:82]: Awesome!

File: content/start/core-concepts/head-and-metadata.mdx [hunk 7]
```diff
@@ -1,17 +1,178 @@
...
+```
+
+### Adding SEO tags
+
+You can add SEO related `Meta` tags either in `root.tsx` or inside your routes. Just like with the `Title`, `Meta` tags used inside a Route component will override the `Meta` tags used in the `Head` component.
+
+```tsx {9-33}
+import { Suspense } from "solid-js";
+import {
+  ...
```
Comment (boehs) [inline content/start/core-concepts/head-and-metadata.mdx:110]: Again, show the meta related imports

File: content/start/core-concepts/routing-and-pages.mdx [hunk 8]
```diff
@@ -14,7 +14,7 @@ Here are a few examples of files in our directory structure and how they would t
 - `/src/routes/admin/index.tsx` -> mysite.com/admin
 - `/src/routes/admin/edit-settings.tsx` -> mysite.com/admin/edit-settings
 
-Under the hood, SolidStart is able to traverse this directory structure, automatically collect all the routes, and populate the `<Routes />` component in the `src/root.tsx` file. This means that all you have to do is create a route in your directory structure and SolidStart takes care of everything else needed to make that route available to visit in your application!
+Under the hood, SolidStart is able to traverse this directory structure, automatically collect all the routes, and give you the `<FileRoutes />` component. You can use this component to specify your routes to the `<Routes />` component in `root.tsx`. This means that all you have to do is create a route in your directory structure and SolidStart takes care of everything else needed to make that route available to visit in your application!
```
Comment (boehs) [inline content/start/core-concepts/routing-and-pages.mdx:17]: Good catch!

File: content/start/core-concepts/routing-and-pages.mdx [hunk 9]
```diff
@@ -50,10 +50,67 @@ And this will render "Welcome to SolidStart, Jane!" In reality, `routeData` will
...
+  Body,
+  ErrorBoundary,
+  FileRoutes,
+  Head,
+  Html,
+  Meta,
+  Routes,
+  Scripts,
+  Title,
+} from "solid-start";
```
Comment (boehs) [inline content/start/core-concepts/routing-and-pages.mdx:68]: Thank you for showing imports!

File: content/start/core-concepts/routing-and-pages.mdx [hunk 10]
```diff
@@ -62,32 +119,41 @@ If you have used Solid App Router in a SolidJS app before, you may notice that t
...
+## Dynamic routes
 
 In some cases, it would be horribly inefficient to create each route manually. When you are dealing with dynamic content, it only makes sense to use dynamic routes!
 
 ### Route parameters
 
 In solid-start, route parameters are defined in the file name. The most basic type of route parameter is enclosed in brackets, like `[id].tsx`. parameters acts as a placeholder for any string. There can be multiple parameters for a single file, but you should ensure there is a delimiter between those parameters.
 
-```
+```ts
```
Comment (boehs) [inline content/start/core-concepts/routing-and-pages.mdx:151]: Not everything is typescript man

**Review by boehs — CHANGES_REQUESTED** [review]

**Comment by boehs** [comment]
Aside from the above changes, this looks good. One other note is that these docs are for a version of solid-start not even *released* yet so like probably wait till .96 before merging

File: content/start/core-concepts/resource-routes.mdx [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+};
+```
+
+## Webhooks
+
+Resource routes can be used to handle webhooks. For example, you can create a webhook that receives notifications from GitHub when a new commit is pushed to a repository:
+
+```ts
+import type { ActionFunction } from "@remix-run/node"; // or cloudflare/deno
+import { json } from "@remix-run/node"; // or cloudflare/deno
```
Comment (boehs) [inline content/start/core-concepts/resource-routes.mdx:111]: Copy paste????

File: content/start/core-concepts/resource-routes.mdx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+      /* handle "PUT" */
+    }
+    case "PATCH": {
+      /* handle "PATCH" */
+    }
+    case "DELETE": {
+      /* handle "DELETE" */
+    }
+  }
+};
```
Comment (boehs) [inline content/start/core-concepts/resource-routes.mdx:102]: The text and example are in conflict

File: content/start/core-concepts/resource-routes.mdx [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+import type { ApiEvent, json } from "solid-start/server";
+
+export const get = async ({ request }: ApiEvent) => {
+  // handle "GET" request
+
+  return json({ success: true }, 200);
+};
+```
+
+To handle `POST`, `PUT`, `PATCH` or `DELETE` requests export an appropriately named function function:
```
Comment (boehs) [inline content/start/core-concepts/resource-routes.mdx:82]:
```suggestion
To handle `POST`, `PUT`, `PATCH` or `DELETE` requests export an appropriately named function:
```

File: content/start/core-concepts/resource-routes.mdx [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+}
+```
+
+When the user clicks the link from the UI route, they will navigate to the PDF.
+
+## Linking to Resource Routes
+
+<docs-error>It’s imperative that you use <code>rel=external</code> on any Links to Resource Routes</docs-error>
+
+There's a subtle detail to be aware of when linking to resource routes. You need to link to it with `<Link reloadDocument>` or a plain `<a href>`. If you link to it with a normal `<NavLink href="report.pdf">` without `rel="external"`, then the resource route will be treated as a UI route and SolidStart will try to do a client-side navigation to that page.
```
Comment (boehs) [inline content/start/core-concepts/resource-routes.mdx:64]:
```suggestion
There's a subtle detail to be aware of when linking to resource routes. You need to link to it with `<Link reloadDocument>` or a plain `<a href>`. If you link to it with a normal `<NavLink href="report.pdf">` without `rel="external"`, then the resource route will be treated as a UI route and SolidStart will try to do client-side navigation to that page.
```

File: content/start/core-concepts/resource-routes.mdx [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+- Creating a PATCH endpoint
+- Creating a `users.json` endpoint
+
+# Resource Routes
+
+Resource Routes are not part of your application UI, but are still part of your application. They can send any kind of Response.
+
+Most routes in SolidStart are UI Routes, or routes that actually render a component. But routes don't always have to render components. There are a handful of cases where you want to use route as a general purpose endpoint to your website. Here are a few examples:
+
+- JSON API for a mobile app that reuses server-side code with the Remix UI
```
Comment (boehs) [inline content/start/core-concepts/resource-routes.mdx:16]: Another copy paste

**Review by boehs — CHANGES_REQUESTED** [review]
I assume this docs is not "finished"

**Comment by Jutanium** [comment]
@nksaraf let's combine this branch and #121

**Comment by boehs** [comment]
Did you really mean to merge this into main?

**Comment by Jutanium** [comment]
@boehs I merged it into another branch, made a bunch of changes, and merged into main

**Pull request merged**